backend-shared/config/db/util/utils_test.go

```diff
@@ -1,12 +1,881 @@
+			Expect(isNew).To(BeTrue())
+
+			// ----------------------------------------------------------------------------
+			By("Verify DeploymentToApplicationMapping resource entry is created in DB.")
+			// ----------------------------------------------------------------------------
+
+			err = dbQueries.GetDeploymentToApplicationMappingByDeplId(ctx, &db.DeploymentToApplicationMapping{
+				Deploymenttoapplicationmapping_uid_id: deploymentToApplicationMapping.Deploymenttoapplicationmapping_uid_id,
+			})
+			Expect(err).To(BeNil())
+
+			// ----------------------------------------------------------------------------
+			By("Try to create same DeploymentToApplicationMapping, it should return existing resource instead of creating new.")
+			// ----------------------------------------------------------------------------
+
+			isNew, err = GetOrCreateDeploymentToApplicationMapping(ctx, deploymentToApplicationMapping, dbQueries, log)
+			Expect(err).To(BeNil())
+			Expect(isNew).To(BeFalse())
+		})
+
+		It("Should delete existing DeploymentToApplicationMapping, if a resource is passed having same name/namespace, so there is be only one resource per name and namespace.", func() {
+
+			// ----------------------------------------------------------------------------
+			By("Create first DeploymentToApplicationMapping resource.")
+			// ----------------------------------------------------------------------------
+
+			isNew, err := GetOrCreateDeploymentToApplicationMapping(ctx, deploymentToApplicationMapping, dbQueries, log)
+			Expect(err).To(BeNil())
+			Expect(isNew).To(BeTrue())
+
+			// ----------------------------------------------------------------------------
+			By("Create second DeploymentToApplicationMapping resource, having same name and namespace as first, but different ID.")
+			// ----------------------------------------------------------------------------
+
+			deploymentToApplicationMappingSecond := &db.DeploymentToApplicationMapping{
+				Deploymenttoapplicationmapping_uid_id: string(uuid.NewUUID()),
+				Application_id:                        application.Application_id,
+				DeploymentName:                        "my-depl-to-app-mapping",
+				DeploymentNamespace:                   "my-namespace",
+				NamespaceUID:                          "test-1",
+			}
+
+			isNew, err = GetOrCreateDeploymentToApplicationMapping(ctx, deploymentToApplicationMappingSecond, dbQueries, log)
+			Expect(err).To(BeNil())
+			Expect(isNew).To(BeTrue())
+
+			// ----------------------------------------------------------------------------
+			By("Check first DeploymentToApplicationMapping is deleted in DB.")
+			// ----------------------------------------------------------------------------
+
+			err = dbQueries.GetDeploymentToApplicationMappingByDeplId(ctx, &db.DeploymentToApplicationMapping{
+				Deploymenttoapplicationmapping_uid_id: deploymentToApplicationMapping.Deploymenttoapplicationmapping_uid_id,
+			})
+			Expect(err).NotTo(BeNil())
```

Comment:
I suggest using `IsResultNotFoundError` on the error, to make sure we are getting the correct error result here.